### PR TITLE
Solved bug of checkbox in student gird.

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**
Solved bug of student correction form through URL.
changed version of simple-react-data-grid.
Added clear search functionality in advance search.

### CODE REVIEW

File: src/components/DataGrid.jsx [hunk 1]
```diff
@@ -102,29 +108,56 @@ class StudentInfoGrid extends Component {
       if (nextProps.students !== this.props.students) {
         this.setState({
           students: this.formattedStudent(nextProps.students),
+          checkedIds: this.allStudentUncheckedInitially(nextProps.students),
         });
+        let tempArray = this.allStudentUncheckedInitially(nextProps.students);
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: tempArray is not a proper name

File: src/components/DataGrid.jsx [hunk 2]
```diff
@@ -102,29 +108,56 @@ class StudentInfoGrid extends Component {
...
+  allStudentUncheckedInitially(students) {
+    return students.map(student => ({ id: student.id, isChecked: false }));
+  }
   refreshStudentsGrid() {
     this.props.getAllStudentsAction({ secretKey: this.props.secretKey });
     this.setState({
       refresh: true,
     });
   }
+  selectedStudents(checkedUncheckedIds) {
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: Incorrect name - checkedUncheckedIds

File: src/components/DataGrid.jsx [hunk 3]
```diff
@@ -146,9 +179,32 @@ class StudentInfoGrid extends Component {
    * @param {Object} selectedRow
    */
   getSelectedRow(selectedRow) {
+    let checkedUncheckedStudentsIds = [];
+    const studentsData = this.state.students.map((student) => {
+      let studentObject = { ...student, id: Number(student.studentId) };
+      selectedRow.forEach((selectedRowStudent) => {
+        if (String(selectedRowStudent.studentId) === String(student.studentId)) {
+          studentObject = { ...student, id: Number(student.studentId), studentId: String(student.studentId), isChecked: selectedRowStudent.isChecked };
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: Break into multiple lines

File: src/components/DataGrid.jsx [hunk 4]
```diff
@@ -146,9 +179,32 @@ class StudentInfoGrid extends Component {
...
+          checkedUncheckedStudentsIds.push({ id: Number(student.studentId), isChecked: selectedRowStudent.isChecked });
+        }
+      });
+      return studentObject;
+    });
+    const tempArray = this.state.checkedIds.map((idObject) => {
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: Incorrect name - tempArray

File: src/components/DataGrid.jsx [hunk 5]
```diff
@@ -146,9 +179,32 @@ class StudentInfoGrid extends Component {
...
+      let finalCheckedUnCheckedIdObj = idObject;
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: Incorrect name - **finalCheckedUnCheckedIdObj** and **idOject**

File: src/components/StudentRegistrationCorrectionFormURL.jsx [hunk 6]
```diff
@@ -738,7 +740,7 @@ class StudentRegistrationCorrectionFormURL extends Component {
 }
 
 StudentRegistrationCorrectionFormURL.propTypes = {
-  studentData: PropTypes.object,
+  studentData: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
```
Comment (pepcus-pratik) [inline src/components/StudentRegistrationCorrectionFormURL.jsx]: How can studentData be of type **String**

File: src/dataGrid.css [hunk 7]
```diff
@@ -722,6 +722,7 @@ html, body {
     box-shadow: 0 1px 1px 0 #e0dbdb;
     -webkit-box-shadow: 0 1px 1px 0 #e0dbdb;
     background-color: #fff;
+    position:relative;
```
Comment (pepcus-pratik) [inline src/dataGrid.css]: Incorrect spacing

File: src/dataGrid.css [hunk 8]
```diff
@@ -1069,8 +1070,17 @@ body,
 .input-radio .input-text {
     min-width: 100%;
     max-width: 100%;
-}
+    display:flex;
+    align-items:center;
 
+}
+.clear-search {
```
Comment (pepcus-pratik) [inline src/dataGrid.css]: Add empty line between each CSS class

File: src/components/DataGrid.jsx [hunk 9]
```diff
@@ -146,9 +179,32 @@ class StudentInfoGrid extends Component {
...
+      checkedUncheckedStudentsIds.forEach((checkedUncheckedStudentIdObject) => {
+        if (Number(idObject.id) === Number(checkedUncheckedStudentIdObject.id)) {
+          finalCheckedUnCheckedIdObj = checkedUncheckedStudentIdObject;
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: **finalCheckedUnCheckedIdObj** should have a proper name

File: src/components/DataGrid.jsx [hunk 10]
```diff
@@ -102,29 +108,56 @@ class StudentInfoGrid extends Component {
...
+  }
   refreshStudentsGrid() {
     this.props.getAllStudentsAction({ secretKey: this.props.secretKey });
     this.setState({
       refresh: true,
     });
   }
+  selectedStudents(checkedUncheckedIds) {
+    let checkedStudents = [];
+    checkedUncheckedIds.forEach((obj) => {
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: `obj` is incorrect. It can have a better name here

File: src/components/DataGrid.jsx [hunk 11]
```diff
@@ -102,29 +108,56 @@ class StudentInfoGrid extends Component {
...
           refresh: false,
+          checkedIds: this.allStudentUncheckedInitially(nextProps.students),
         });
+        let tempArray = this.allStudentUncheckedInitially(nextProps.students);
+        this.selectedStudents(tempArray);
       }
     }
   }
-
+  allStudentUncheckedInitially(students) {
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: Incorrect method name

File: src/components/DataGrid.jsx [hunk 12]
```diff
@@ -75,6 +76,8 @@ class StudentInfoGrid extends Component {
     this.formattedStudent = this.formattedStudent.bind(this);
     this.getSelectedRow = this.getSelectedRow.bind(this);
     this.refreshStudentsGrid = this.refreshStudentsGrid.bind(this);
+    this.allStudentUncheckedInitially = this.allStudentUncheckedInitially.bind(this);
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: try starting the method name with `verb`

File: src/components/DataGrid.jsx [hunk 13]
```diff
@@ -75,6 +76,8 @@ class StudentInfoGrid extends Component {
...
+    this.selectedStudents = this.selectedStudents.bind(this);
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: Can it be named **getSelectedStudents**??

File: src/components/DataGrid.jsx [hunk 14]
```diff
@@ -314,8 +370,12 @@ class StudentInfoGrid extends Component {
    * @return {Array} students
    */
   formattedStudent(students) {
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: Rename this method to `formatStudents`

File: src/components/DataGrid.jsx (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: I have renamed it by idIsCheckedStatusList

File: src/components/DataGrid.jsx (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Solved.

File: src/components/DataGrid.jsx (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Done.

File: src/components/DataGrid.jsx (hunk 4 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: I have renamed it by idIsCheckedStatusList.

File: src/components/DataGrid.jsx (hunk 5 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: I have renamed it by finalIdIsCheckedStatusListObject .

File: src/components/DataGrid.jsx (hunk 10 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Done.

File: src/components/DataGrid.jsx (hunk 11 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: I have renamed it by setStudentsISUncheckedInitially.

File: src/components/DataGrid.jsx (hunk 12 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Done.

File: src/components/DataGrid.jsx (hunk 13 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Done.

File: src/components/DataGrid.jsx (hunk 14 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Done.

**Pull request merged**